cmd/cli/license/inspect.go

```diff
@@ -0,0 +1,207 @@
+	}{
+		Alias:    Alias(j),
+		Metadata: j.Metadata,
+	}
+	return json.Marshal(data)
+}
+
+// MarshalYAML implements custom YAML marshaling
+func (j jsonOutput) MarshalYAML() (interface{}, error) {
+	type Alias jsonOutput // prevent recursion
+	data := struct {
+		Alias
+		Metadata map[string]string `yaml:"Metadata"`
+	}{
+		Alias:    Alias(j),
+		Metadata: j.Metadata,
+	}
+	return data, nil
+}
+
+// Add this struct after the LicenseInfo struct
+type licenseFile struct {
+	License string `json:"license"`
+}
+
+// Update table columns
+var licenseProductColumn = output.TableColumn[LicenseInfo]{
+	ColumnConfig: table.ColumnConfig{Name: "PRODUCT"},
+	Value:        func(l LicenseInfo) string { return l.Product },
+}
+
+var licenseLicenseIDColumn = output.TableColumn[LicenseInfo]{
+	ColumnConfig: table.ColumnConfig{Name: "LICENSE ID"},
+	Value:        func(l LicenseInfo) string { return l.LicenseID },
+}
+
+var licenseCustomerIDColumn = output.TableColumn[LicenseInfo]{
+	ColumnConfig: table.ColumnConfig{Name: "CUSTOMER ID"},
+	Value:        func(l LicenseInfo) string { return l.CustomerID },
+}
+
+var licenseValidUntilColumn = output.TableColumn[LicenseInfo]{
+	ColumnConfig: table.ColumnConfig{Name: "VALID UNTIL"},
+	Value:        func(l LicenseInfo) string { return l.ValidUntil },
+}
+
+var licenseLicenseVersionColumn = output.TableColumn[LicenseInfo]{
+	ColumnConfig: table.ColumnConfig{Name: "VERSION"},
+	Value:        func(l LicenseInfo) string { return l.LicenseVersion },
+}
+
+var licenseCapabilitiesColumn = output.TableColumn[LicenseInfo]{
+	ColumnConfig: table.ColumnConfig{Name: "CAPABILITIES"},
+	Value: func(l LicenseInfo) string {
+		if len(l.Capabilities) == 0 {
+			return "none"
+		}
+		var caps []string
+		for k, v := range l.Capabilities {
+			caps = append(caps, fmt.Sprintf("%s=%s", k, v))
+		}
+		return strings.Join(caps, ", ")
+	},
+}
+
+func (o *InspectOptions) Run(ctx context.Context, cmd *cobra.Command) error {
+	// Read the license file
+	data, err := os.ReadFile(o.LicenseFile)
+	if err != nil {
+		return fmt.Errorf("failed to read license file: %w", err)
+	}
+
+	// Parse the license file
+	var license licenseFile
+	if err := json.Unmarshal(data, &license); err != nil {
+		return fmt.Errorf("failed to parse license file: %w", err)
+	}
+
+	// Create offline license validator
+	validator, err := licensepkg.NewOfflineLicenseValidator()
+	if err != nil {
+		return fmt.Errorf("failed to create license validator: %w", err)
+	}
+
+	// Validate the license token
+	claims, err := validator.ValidateToken(license.License)
+	if err != nil {
+		return fmt.Errorf("invalid license: %w", err)
+	}
+
+	// Create license info from the validated claims
+	licenseInfo := LicenseInfo{
+		Product:        claims.Product,
+		LicenseID:      claims.LicenseID,
+		CustomerID:     claims.CustomerID,
+		ValidUntil:     claims.ExpiresAt.Format(time.DateOnly),
+		LicenseVersion: claims.LicenseVersion,
+		Capabilities:   claims.Capabilities,
+	}
```

Comment:
can we just print out the claims type? I understand the value of separating types, but half of this code here is because of the separation and having to do custom unmarshalling logic, and I don't see the value at this point. Lets start simple

Reply:
@wdbaruni I want to separate the presentation layer from what we get in the actual license token claims.
Currently, they are similar, but it is a good practice to just separate them.

Unless there is something wrong with it that I do not see, I would prefer to keep it.

Reply:
Directly using claims when the output has changed